File: results-05-12/typescript/typescript-00015-benchmark-pdf-store/qwen2-5-coder-32b/editor/final-src-code.ts

  // These environment variables are required, before any import.
  // Do not remove them, as they set environment variables for the Shinkai Tools.
  Deno.env.set('SHINKAI_NODE_LOCATION', "http://localhost:9950");
  Deno.env.set('BEARER', "debug");
  Deno.env.set('X_SHINKAI_TOOL_ID', "tool-id-debug");
  Deno.env.set('X_SHINKAI_APP_ID', "tool-app-debug");
  Deno.env.set('X_SHINKAI_LLM_PROVIDER', "o_qwen2_5_coder_32b");
  Deno.env.set('HOME', "results/typescript/typescript-00015-benchmark-pdf-store/qwen2-5-coder-32b/editor/home");
  Deno.env.set('MOUNT', "results/typescript/typescript-00015-benchmark-pdf-store/qwen2-5-coder-32b/editor/mount");
  Deno.env.set('ASSETS', "results/typescript/typescript-00015-benchmark-pdf-store/qwen2-5-coder-32b/editor/assets");
  
import { getHomePath } from './shinkai-local-support.ts';
import { fetch } from './globalthis/implementation'; // To ensure fetch is available
import { PDFDocument } from 'npm:@pdf-lib/pdf-lib';

type CONFIG = {};
type INPUTS = { url: string };
type OUTPUT = { filePath: string };

export async function run(config: CONFIG, inputs: INPUTS): Promise<OUTPUT> {
    const response = await fetch(inputs.url);
    if (!response.ok) {
        throw new Error(`Failed to download PDF from ${inputs.url}: ${response.statusText}`);
    }
    
    const arrayBuffer = await response.arrayBuffer();
    const pdfBytes = new Uint8Array(arrayBuffer);

    // Parse the PDF using @pdf-lib/pdf-lib
    const pdfDoc = await PDFDocument.load(pdfBytes);
    const pages = pdfDoc.getPages();

    let textContent = '';
    for (const page of pages) {
        const content = await page.getTextContent();
        textContent += content.map(item => item.str).join(' ') + '\n';
    }

    const homePath = getHomePath();
    const filePath = `${homePath}/downloaded-pdf.txt`;

    await Deno.writeFile(filePath, new TextEncoder().encode(textContent));

    return { filePath };
}

  
  // console.log('Running...')
  // console.log('Config: {}')
  // console.log('Inputs: {"url":"https://raw.githubusercontent.com/acedward/expert-octo-computing-machine/main/sample.pdf"}')
  
  try {
    const program_result = await run({}, {"url":"https://raw.githubusercontent.com/acedward/expert-octo-computing-machine/main/sample.pdf"});
    if (program_result) console.log(JSON.stringify(program_result, null, 2));
    else console.log(program_result);
  } catch (e) {
    console.log('::ERROR::', e);
  }

File: results-05-12/python/python-00005-benchmark-file-store/qwen2-5-coder-32b/editor/final-src-code.py

import os
import json
os.environ["SHINKAI_NODE_LOCATION"] = "http://localhost:9950"
os.environ["BEARER"] = "debug"
os.environ["X_SHINKAI_TOOL_ID"] = "tool-id-debug"
os.environ["X_SHINKAI_APP_ID"] = "tool-app-debug"
os.environ["X_SHINKAI_LLM_PROVIDER"] = "o_qwen2_5_coder_32b"
os.environ["HOME"] = "results/python/python-00005-benchmark-file-store/qwen2-5-coder-32b/editor/home"
os.environ["MOUNT"] = "results/python/python-00005-benchmark-file-store/qwen2-5-coder-32b/editor/mount"
os.environ["ASSETS"] = "results/python/python-00005-benchmark-file-store/qwen2-5-coder-32b/editor/assets"

from typing import Optional
import requests
from shinkai_local_support import get_home_path

class CONFIG:
    pass

class INPUTS:
    url: str

class OUTPUT:
    file_path: Optional[str]

async def run(config: CONFIG, inputs: INPUTS) -> OUTPUT:
    try:
        response = requests.get(inputs.url)
        response.raise_for_status()
        
        home_path = get_home_path()
        file_name = inputs.url.split('/')[-1]
        file_path = f"{home_path}/{file_name}"
        
        with open(file_path, 'wb') as file:
            file.write(response.content)
            
        output = OUTPUT()
        output.file_path = file_path
        return output
    
    except Exception as e:
        output = OUTPUT()
        output.file_path = None  # or handle the exception as needed
        return output

if __name__ == "__main__":
    import asyncio
    
    config = CONFIG()
    inputs = INPUTS()
    inputs.url="https://raw.githubusercontent.com/acedward/expert-octo-computing-machine/main/screenshot.png"
    
    # Run the async function
    result = asyncio.run(run(config, inputs))
    print(json.dumps(result.__dict__))
